scenariorun/src/main/kotlin/com/cosmotech/scenariorun/azure/ScenarioRunServiceImpl.kt

```diff
@@ -102,7 +102,21 @@ internal class ScenarioRunServiceImpl(
       // TODO Only the owner or an admin should be able to perform this operation
       throw CsmAccessForbiddenException("You are not allowed to delete this Resource")
     }
-    this.deleteScenarioRunWithoutAccessEnforcement(scenarioRun)
+    deleteScenarioRunWithoutAccessEnforcement(scenarioRun)
+    deleteScenarioMetadataWithoutAccessEnforcement(scenarioRun)
```

Comment:
This code and corresponding method should be in ScenarioServiceImpl.kt.
Here it will delete all scenariometadata if 1 scenario run is deleted.

Reply:
As scenario does not contains an instance of scenariorun and use event that why I kept this 
Your right we should delete metata only when deleting a scenario